salt/modules/pip.py

```diff
@@ -120,37 +122,58 @@ def _get_pip_bin(bin_env):
     executable itself, or from searching conventional filesystem locations
     '''
     if not bin_env:
-        which_result = __salt__['cmd.which_bin'](
-            ['pip{0}.{1}'.format(*sys.version_info[:2]),
-             'pip{0}'.format(sys.version_info[0]),
-             'pip', 'pip-python']
-        )
-        if salt.utils.is_windows() and six.PY2:
-            which_result.encode('string-escape')
-        if which_result is None:
-            raise CommandNotFoundError('Could not find a `pip` binary')
-        return which_result
+        logger.debug('PIP: Using pip from calling python')
+        return [os.path.normpath(sys.executable), '-m', 'pip']
+
+    # try to get python bin from virtualenv, bin_env
+    python_bin = 'python'
+    if salt.utils.is_windows():
```

Comment:
How about:
`python_bin = 'python.exe' if salt.utils.is_windows() else 'python`?
  It saves two source lines, eliminates an unneeded assignment (on Windows)
and is easier to read and understand.